app/controllers/users_controller.rb

```diff
@@ -0,0 +1,20 @@
+class UsersController < ApplicationController
+    get '/users/new' do
+        erb :"users/new.html"
+    end
+
+    post '/users' do
+        user = User.new(email: params[:email], password: params[:password], password_confirmation: params[:password_confirmation])
+        if user.save
+            flash[:notice] = "Thanks for signing up! You may now log in!"
+            redirect "/sessions/new"
+        else
+            # for allowing the email to be retained across pages
+            @email = params[:email]
```

Comment:
Rather than storing the email they put in the form as an instance variable, you can pass the `user` object into the `erb` template via. `locals`